### PR TITLE
Start writing E1.37-2 RDM tests

### PR DESCRIPTION
**Pull request by peternewman**

### CODE REVIEW

File: tools/rdm/TestDefinitions.py [hunk 1]
```diff
@@ -680,6 +680,17 @@ class GetSupportedParameters(ResponderTestFixture):
       ('LOCK_STATE', 'LOCK_STATE_DESCRIPTION'),
   ]
 
+  # If any PID apart from the first in each group is supported, the first one
+  # must be.
+  PID_REVERSE_DEPENDENCIES = [
```
Comment (nomis52) [inline tools/rdm/TestDefinitions.py]: I'd make this a list of tuples in the form:

[  
  ( first_pid, [other_pids]),
]

File: tools/rdm/TestDefinitions.py [hunk 2]
```diff
@@ -777,6 +788,24 @@ def VerifyResult(self, response, fields):
         self.AddAdvisory('%s supported but %s is not' %
                          (pid_names[0], ','.join(unsupported_pids)))
 
+    for pid_names in self.PID_REVERSE_DEPENDENCIES:
```
Comment (nomis52) [inline tools/rdm/TestDefinitions.py]: then you can write this as

for required_pid, other_pids in self.PID_REVERSE_DEPENDECIES:

File: tools/rdm/TestDefinitions.py [hunk 3]
```diff
@@ -777,6 +788,24 @@ def VerifyResult(self, response, fields):
...
+      if self.LookupPid(pid_names[0]).value in supported_parameters:
+        continue
+
+      dependant_pids = []
+      for pid_name in pid_names[1:]:
+        pid = self.LookupPid(pid_name)
+        if pid is None:
+          self.SetBroken('Missing PID %s' % pid_name)
```
Comment (nomis52) [inline tools/rdm/TestDefinitions.py]: Hmm, this seems wrong. Broken is supposed to indicate a code error. I think this (and the one above) should be SetFailed() 

See https://www.openlighting.org/rdm-tools/rdm-responder-tests/interpreting-the-test-output/

File: tools/rdm/TestDefinitions.py [hunk 4]
```diff
@@ -6403,6 +6432,50 @@ class AllSubDevicesGetPresetMergeMode(TestMixins.AllSubDevicesGetMixin,
...
+    if not response.WasAcked():
+      self.SetProperty('list_interfaces', [])
+      return
+
+    interfaces = []
+
+    for interface in fields['interfaces']:
+      interface_id = interface['interface_identifier']
+      interfaces.append(interface_id)
+      if interface['interface_hardware_type'] != 1:
```
Comment (nomis52) [inline tools/rdm/TestDefinitions.py]: Make 1 an enum ?

File: tools/rdm/TestDefinitions.py (hunk 3 above)
Comment (peternewman) [inline tools/rdm/TestDefinitions.py]: Yes, I think that's correct; it's just doing PID name (in the code) to object lookup here, if that fails it is a code error.

File: tools/rdm/TestDefinitions.py (hunk 1 above)
Comment (peternewman) [inline tools/rdm/TestDefinitions.py]: Done, and for PID_DEPENDENCIES

File: tools/rdm/TestDefinitions.py (hunk 2 above)
Comment (peternewman) [inline tools/rdm/TestDefinitions.py]: Done

File: tools/rdm/TestDefinitions.py (hunk 3 above)
Comment (peternewman) [inline tools/rdm/TestDefinitions.py]: I've clarified the error message too.

File: tools/rdm/TestDefinitions.py (hunk 4 above)
Comment (peternewman) [inline tools/rdm/TestDefinitions.py]: Done.

**Pull request merged**